# Project 2: Karamoja Uganda
### History:
he Karamoja sub-region, commonly known as Karamoja, is a region in Uganda.The region is projected to have a population of 1.4 millions in 2022 by UBOS. The region was ruled by the British from 1916 to 1962.The famine that struck the region in 1980 was, in terms of mortality rates one of the worst in history. 21% of the population died, including 60% of the infants.

### For this project:
we will clean up the data for this project
analyse the data associated with the current famine
come up with solutions to mitigate the famine

### Cleaning up the data

In [1]:
import pandas as pd
#loading the files drom the data folder
with open(r"C:\Users\ericg\Downloads\DATA\DATA\TABLES\Uganda_Karamoja_District_Crop_Yield_Population.csv") as a:
    df_a = pd.read_csv(a)
df_a
# print(df.head())


OBJECTID           NAME     POP        Area  S_Yield_Ha  M_Yield_Ha  \
0        92           ABIM   90385  2771977106         449        1040   
1        96         AMUDAT  101790  1643582836         205        1297   
2        20        KAABONG  627057  7373606003         279         945   
3        85         KOTIDO  243157  3641539808         331        1148   
4         5         MOROTO  127811  3570160948         128         355   
5        54  NAKAPIRIPIRIT  146780  4216323900         356        1264   
6        80          NAPAK  167625  4508782023         137         854   

   Crop_Area_Ha     S_Area_Ha    M_Area_Ha  S_Prod_Tot  M_Prod_Tot  
0   5470.068394   3277.295971  1848.621855     1471506     1922567  
1   5765.443719   2973.423860  2733.661014      609552     3545558  
2  28121.672530  20544.194960  7394.416334     5731830     6987723  
3  53032.649450  50247.443900  1751.372284    16631904     2010575  
4   5954.814048   4741.748776  1190.050606      606944      422468  
5  26372.698490  19237.333210  6425.788414     6848491     8122197  
6  22944.296020  16142.015880  6543.719066     2211456     5588336

In [2]:
df_a.shape

(7, 11)

In [3]:
#loading the files
with open(r"C:\Users\ericg\Downloads\DATA\DATA\TABLES\Uganda_Karamoja_Subcounty_Crop_Yield_Population.csv")as b:
    df_b = pd.read_csv(b)
df_b

OBJECTID              SUBCOUNTY_NAME  DISTRICT_NAME     POP        Area  \
0        263                     KACHERI         KOTIDO   17244  1067176155   
1        264                      KOTIDO         KOTIDO   52771   597575188   
2        265         KOTIDO TOWN COUNCIL         KOTIDO   27389    23972401   
3        266                NAKAPERIMORU         KOTIDO   38775   419111591   
4        267                  PANYANGARA         KOTIDO   65704   880955930   
5        268                      RENGEN         KOTIDO   41273   652744859   
6        591               KAABONG  EAST        KAABONG   42221    60801942   
7        592        KAABONG TOWN COUNCIL        KAABONG   38857    13071455   
8        593                KAABONG WEST        KAABONG   41454    67612362   
9        594                    KALAPATA        KAABONG   99203   223116860   
10       595                      KAMION        KAABONG   60070  1199409465   
11       596                      KAPEDO        KAABONG   33951   304557831   
12       597                     KATHILE        KAABONG   56870   228457417   
13       598                   KAWALAKOL        KAABONG   37691  1742041261   
14       599                  LOBALANGIT        KAABONG   22907   302572789   
15       600                      LODIKO        KAABONG   31820   397700845   
16       601                     LOLELIA        KAABONG  100919  1069331387   
17       602                      LOYORO        KAABONG   16608  1330019236   
18       603            NAPORE (KARENGA)        KAABONG   20394   309577387   
19       604                       SIDOK        KAABONG   24092   125330583   
20       765                        ABIM           ABIM   14377   188613057   
21       766           ABIM TOWN COUNCIL           ABIM   11489   115941152   
22       767                      ALEREK           ABIM   12573  1255760525   
23       768                     LOTUKEI           ABIM   18641   191684465   
24       769                     MORULEM           ABIM   16834   250364895   
25       810                      KARITA         AMUDAT   45015   658008801   
26       811                       LOROO         AMUDAT   22541   371970302   
27       812                      AMUDAT         AMUDAT   30181   596393139   
28       813         AMUDAT TOWN COUNCIL         AMUDAT    4053    17207110   
29       925                  KATIKEKILE         MOROTO   11772   277451997   
30       926                    NADUNGET         MOROTO   42115   596023780   
31       927                       TAPAC         MOROTO   21999   622266516   
32       928           NORTHERN DIVISION         MOROTO    7161     2738777   
33       929           SOUTHERN DIVISION         MOROTO    3272     2121209   
34       978                     LOREGAE  NAKAPIRIPIRIT   26644   928683431   
35       979  NAKAPIRIPIRIT TOWN COUNCIL  NAKAPIRIPIRIT    1418     4673094   
36       980                      NAMALU  NAKAPIRIPIRIT   25905   553067364   
37       981                 KAKOMONGOLE  NAKAPIRIPIRIT   15960   312709097   
38       982                    LOLACHAT  NAKAPIRIPIRIT   22801   642401705   
39       983                 LORENGEDWAT  NAKAPIRIPIRIT    9929   435714604   
40       984                   NABILATUK  NAKAPIRIPIRIT   27534   499776866   
41      1023                      IRIIRI          NAPAK   46979  1030623258   
42      1024                      LOPEEI          NAPAK   18815   790782350   
43      1025                 LORENGECORA          NAPAK    2503   118997955   
44      1026                      LOTOME          NAPAK   17834   125857565   
45      1027                      MATANY          NAPAK   29495   338252511   
46      1028          NAPAK TOWN COUNCIL          NAPAK    8010   150906375   
47      1029                   NGOLERIET          NAPAK   20790   158887552   
48      1295                     NYAKWAE           ABIM   16470   769609687   
49      1313                      LOKOPO          NAPA

In [4]:
df_b.shape

(52, 13)

In [5]:
#it seems that the data set is split into different folders and tables hence i have to combine the data and do 'concantination'
#data frame a and b have different collumn numbers

In [6]:
data_frames = [df_a, df_b]  #creating them into a list to make concatation easier
concated_data_frame = pd.concat(data_frames)
concated_data_frame #the joined dataframe

OBJECTID           NAME     POP        Area  S_Yield_Ha   M_Yield_Ha  \
0         92           ABIM   90385  2771977106  449.000000  1040.000000   
1         96         AMUDAT  101790  1643582836  205.000000  1297.000000   
2         20        KAABONG  627057  7373606003  279.000000   945.000000   
3         85         KOTIDO  243157  3641539808  331.000000  1148.000000   
4          5         MOROTO  127811  3570160948  128.000000   355.000000   
5         54  NAKAPIRIPIRIT  146780  4216323900  356.000000  1264.000000   
6         80          NAPAK  167625  4508782023  137.000000   854.000000   
0        263            NaN   17244  1067176155  354.207411  1137.467019   
1        264            NaN   52771   597575188  367.890523  1162.996687   
2        265            NaN   27389    23972401  369.314177  1167.005832   
3        266            NaN   38775   419111591  283.324569   852.366578   
4        267            NaN   65704   880955930  373.836926  1283.859882   
5        268            NaN   41273   652744859  271.185843  1047.180128   
6        591            NaN   42221    60801942  160.588525   650.123565   
7        592            NaN   38857    13071455  212.210703  1229.213444   
8        593            NaN   41454    67612362  208.040518  1056.416068   
9        594            NaN   99203   223116860  209.809352   633.423182   
10       595            NaN   60070  1199409465  259.898246   714.475743   
11       596            NaN   33951   304557831  301.154328  1030.064093   
12       597            NaN   56870   228457417  312.278615  1125.685462   
13       598            NaN   37691  1742041261  298.449959   871.612802   
14       599            NaN   22907   302572789  386.529819   985.192699   
15       600            NaN   31820   397700845  164.652061   815.255729   
16       601            NaN  100919  1069331387  285.213427  1273.397362   
17       602            NaN   16608  1330019236  284.183058  1303.403556   
18       603            NaN   20394   309577387  295.734350   765.334925   
19       604            NaN   24092   125330583  285.336294  1396.991494   
20       765            NaN   14377   188613057  501.263249  1183.089648   
21       766            NaN   11489   115941152  439.750995  1061.151087   
22       767            NaN   12573  1255760525  520.407672  1259.122045   
23       768            NaN   18641   191684465  382.868986  1003.303910   
24       769            NaN   16834   250364895  407.500417   962.958170   
25       810            NaN   45015   658008801  193.904842  1286.723173   
26       811            NaN   22541   371970302  200.696210  1196.088040   
27       812            NaN   30181   596393139  239.047878  1379.045945   
28       813            NaN    4053    17207110  146.117797   861.208491   
29       925            NaN   11772   277451997  311.006215   546.140294   
30       926            NaN   42115   596023780  133.667936   591.270355   
31       927            NaN   21999   622266516  155.869934   211.965635   
32       928            NaN    7161     2738777  131.977698   241.538427   
33       929            NaN    3272     2121209  114.795569     0.000000   
34       978            NaN   26644   928683431  384.256293  1114.593607   
35       979            NaN    1418     4673094  482.438793  1257.802716   
36       980            NaN   25905   553067364  560.313070  1325.254872   
37       981            NaN   15960   312709097  398.979005  1318.261936   
38       982            NaN   22801   642401705  334.284302  1225.930541   
39       983            NaN    9929   435714604  188.699612   795.733280   
40       984            NaN   27534   499776866  259.454122  1066.204758   
41      1023            NaN   46979  1030623258  229.299157   873.452241   
42      1024            NaN   18815   790782350  108.156411  1053.257168   
43      1025            NaN    2503   118997955  184.081197   841.815011   
44      1026            NaN   17834   125857565  128

In [7]:
#cleaning up the data by reducing the number of decimal places tot two 
 #converting the dataframe to intagers as the round function does not apply to strings
numeric_cols = concated_data_frame.select_dtypes(include=['number']).columns
#rounding up the data in the dataframe
concated_data_frame[numeric_cols] = concated_data_frame[numeric_cols].apply(lambda x: round(x, 2))
#returning the first five rows in the dataframe
concated_data_frame.head()

OBJECTID     NAME     POP        Area  S_Yield_Ha  M_Yield_Ha  \
0        92     ABIM   90385  2771977106       449.0      1040.0   
1        96   AMUDAT  101790  1643582836       205.0      1297.0   
2        20  KAABONG  627057  7373606003       279.0       945.0   
3        85   KOTIDO  243157  3641539808       331.0      1148.0   
4         5   MOROTO  127811  3570160948       128.0       355.0   

   Crop_Area_Ha  S_Area_Ha  M_Area_Ha  S_Prod_Tot  M_Prod_Tot SUBCOUNTY_NAME  \
0       5470.07    3277.30    1848.62   1471506.0   1922567.0            NaN   
1       5765.44    2973.42    2733.66    609552.0   3545558.0            NaN   
2      28121.67   20544.19    7394.42   5731830.0   6987723.0            NaN   
3      53032.65   50247.44    1751.37  16631904.0   2010575.0            NaN   
4       5954.81    4741.75    1190.05    606944.0    422468.0            NaN   

  DISTRICT_NAME Karamoja  
0           NaN      NaN  
1           NaN      NaN  
2           NaN      NaN  
3           NaN      NaN  
4           NaN      NaN

In [8]:
#'NAME', collumns mostly seem to be empty
#SUBCOUNTY_NAME, DISTRICT_NAME, Karamoja  apear to have missing values 

In [9]:
# confirming column names
concated_data_frame.columns


Index(['OBJECTID', 'NAME', 'POP', 'Area', 'S_Yield_Ha', 'M_Yield_Ha',
       'Crop_Area_Ha', 'S_Area_Ha', 'M_Area_Ha', 'S_Prod_Tot', 'M_Prod_Tot',
       'SUBCOUNTY_NAME', 'DISTRICT_NAME', 'Karamoja'],
      dtype='object')

In [10]:
#confirming the shape of the dataframe
concated_data_frame.shape


(59, 14)

In [11]:
#lets arrange the dataframe in ascending order with refrence to 'object id'
sorted_concat_df = concated_data_frame.sort_values(by='OBJECTID', ascending=True)
sorted_concat_df

#we are sorting the data to deteramin if there are missing valuse according to 'OBJECTID' 

OBJECTID           NAME     POP        Area  S_Yield_Ha  M_Yield_Ha  \
4          5         MOROTO  127811  3570160948      128.00      355.00   
2         20        KAABONG  627057  7373606003      279.00      945.00   
5         54  NAKAPIRIPIRIT  146780  4216323900      356.00     1264.00   
6         80          NAPAK  167625  4508782023      137.00      854.00   
3         85         KOTIDO  243157  3641539808      331.00     1148.00   
0         92           ABIM   90385  2771977106      449.00     1040.00   
1         96         AMUDAT  101790  1643582836      205.00     1297.00   
0        263            NaN   17244  1067176155      354.21     1137.47   
1        264            NaN   52771   597575188      367.89     1163.00   
2        265            NaN   27389    23972401      369.31     1167.01   
3        266            NaN   38775   419111591      283.32      852.37   
4        267            NaN   65704   880955930      373.84     1283.86   
5        268            NaN   41273   652744859      271.19     1047.18   
6        591            NaN   42221    60801942      160.59      650.12   
7        592            NaN   38857    13071455      212.21     1229.21   
8        593            NaN   41454    67612362      208.04     1056.42   
9        594            NaN   99203   223116860      209.81      633.42   
10       595            NaN   60070  1199409465      259.90      714.48   
11       596            NaN   33951   304557831      301.15     1030.06   
12       597            NaN   56870   228457417      312.28     1125.69   
13       598            NaN   37691  1742041261      298.45      871.61   
14       599            NaN   22907   302572789      386.53      985.19   
15       600            NaN   31820   397700845      164.65      815.26   
16       601            NaN  100919  1069331387      285.21     1273.40   
17       602            NaN   16608  1330019236      284.18     1303.40   
18       603            NaN   20394   309577387      295.73      765.33   
19       604            NaN   24092   125330583      285.34     1396.99   
20       765            NaN   14377   188613057      501.26     1183.09   
21       766            NaN   11489   115941152      439.75     1061.15   
22       767            NaN   12573  1255760525      520.41     1259.12   
23       768            NaN   18641   191684465      382.87     1003.30   
24       769            NaN   16834   250364895      407.50      962.96   
25       810            NaN   45015   658008801      193.90     1286.72   
26       811            NaN   22541   371970302      200.70     1196.09   
27       812            NaN   30181   596393139      239.05     1379.05   
28       813            NaN    4053    17207110      146.12      861.21   
29       925            NaN   11772   277451997      311.01      546.14   
30       926            NaN   42115   596023780      133.67      591.27   
31       927            NaN   21999   622266516      155.87      211.97   
32       928            NaN    7161     2738777      131.98      241.54   
33       929            NaN    3272     2121209      114.80        0.00   
34       978            NaN   26644   928683431      384.26     1114.59   
35       979            NaN    1418     4673094      482.44     1257.80   
36       980            NaN   25905   553067364      560.31     1325.25   
37       981            NaN   15960   312709097      398.98     1318.26   
38       982            NaN   22801   642401705      334.28     1225.93   
39       983            NaN    9929   435714604      188.70      795.73   
40       984            NaN   27534   499776866      259.45     1066.20   
41      1023            NaN   46979  1030623258      229.30      873.45   
42      1024            NaN   18815   790782350      108.16     1053.26   
43      1025            NaN    2503   118997955      184.08      841.82   
44      1026            NaN   17834   125857565      128.56      507.91   
45      1027            NaN  

In [12]:
#how many subcounties are in the dataset
print(sorted_concat_df['SUBCOUNTY_NAME'].unique())
#how many subcounties are in the dataset
print(f"There are {len(sorted_concat_df['SUBCOUNTY_NAME'].unique())} subcounties")

[nan 'KACHERI' 'KOTIDO' 'KOTIDO TOWN COUNCIL' 'NAKAPERIMORU' 'PANYANGARA'
 'RENGEN' 'KAABONG  EAST' 'KAABONG TOWN COUNCIL' 'KAABONG WEST' 'KALAPATA'
 'KAMION' 'KAPEDO' 'KATHILE' 'KAWALAKOL' 'LOBALANGIT' 'LODIKO' 'LOLELIA'
 'LOYORO' 'NAPORE (KARENGA)' 'SIDOK' 'ABIM' 'ABIM TOWN COUNCIL' 'ALEREK'
 'LOTUKEI' 'MORULEM' 'KARITA' 'LOROO' 'AMUDAT' 'AMUDAT TOWN COUNCIL'
 'KATIKEKILE' 'NADUNGET' 'TAPAC' 'NORTHERN DIVISION' 'SOUTHERN DIVISION'
 'LOREGAE' 'NAKAPIRIPIRIT TOWN COUNCIL' 'NAMALU' 'KAKOMONGOLE' 'LOLACHAT'
 'LORENGEDWAT' 'NABILATUK' 'IRIIRI' 'LOPEEI' 'LORENGECORA' 'LOTOME'
 'MATANY' 'NAPAK TOWN COUNCIL' 'NGOLERIET' 'NYAKWAE' 'LOKOPO' 'RUPA'
 'MORUITA']
There are 53 subcounties


In [13]:
#how many district names are in the dataset
print(sorted_concat_df['DISTRICT_NAME'].unique())
#how many districts are in the dataset
print(f"There are {len(sorted_concat_df['DISTRICT_NAME'].unique())} districts")

[nan 'KOTIDO' 'KAABONG' 'ABIM' 'AMUDAT' 'MOROTO' 'NAKAPIRIPIRIT' 'NAPAK']
There are 8 districts


In [14]:
#the OBJECTID column seems to be missing data

In [15]:
# Drop the column 'NAME' from the DataFrame as most of the cells are empty
sorted_concat_df = sorted_concat_df.drop(columns=['NAME'])
sorted_concat_df



OBJECTID     POP        Area  S_Yield_Ha  M_Yield_Ha  Crop_Area_Ha  \
4          5  127811  3570160948      128.00      355.00       5954.81   
2         20  627057  7373606003      279.00      945.00      28121.67   
5         54  146780  4216323900      356.00     1264.00      26372.70   
6         80  167625  4508782023      137.00      854.00      22944.30   
3         85  243157  3641539808      331.00     1148.00      53032.65   
0         92   90385  2771977106      449.00     1040.00       5470.07   
1         96  101790  1643582836      205.00     1297.00       5765.44   
0        263   17244  1067176155      354.21     1137.47       7023.53   
1        264   52771   597575188      367.89     1163.00      13587.99   
2        265   27389    23972401      369.31     1167.01       1656.53   
3        266   38775   419111591      283.32      852.37       7087.82   
4        267   65704   880955930      373.84     1283.86      10398.25   
5        268   41273   652744859      271.19     1047.18      13278.52   
6        591   42221    60801942      160.59      650.12       1021.32   
7        592   38857    13071455      212.21     1229.21        175.08   
8        593   41454    67612362      208.04     1056.42        967.34   
9        594   99203   223116860      209.81      633.42       2834.35   
10       595   60070  1199409465      259.90      714.48        997.16   
11       596   33951   304557831      301.15     1030.06       2365.26   
12       597   56870   228457417      312.28     1125.69       6890.70   
13       598   37691  1742041261      298.45      871.61       3934.43   
14       599   22907   302572789      386.53      985.19       1890.25   
15       600   31820   397700845      164.65      815.26        401.80   
16       601  100919  1069331387      285.21     1273.40       2700.68   
17       602   16608  1330019236      284.18     1303.40       1491.77   
18       603   20394   309577387      295.73      765.33       1494.05   
19       604   24092   125330583      285.34     1396.99        957.49   
20       765   14377   188613057      501.26     1183.09        999.37   
21       766   11489   115941152      439.75     1061.15        788.46   
22       767   12573  1255760525      520.41     1259.12       1266.11   
23       768   18641   191684465      382.87     1003.30        698.17   
24       769   16834   250364895      407.50      962.96        925.05   
25       810   45015   658008801      193.90     1286.72       2513.27   
26       811   22541   371970302      200.70     1196.09       1539.40   
27       812   30181   596393139      239.05     1379.05       1652.00   
28       813    4053    17207110      146.12      861.21         60.77   
29       925   11772   277451997      311.01      546.14        145.24   
30       926   42115   596023780      133.67      591.27       2480.12   
31       927   21999   622266516      155.87      211.97       1111.66   
32       928    7161     2738777      131.98      241.54          0.32   
33       929    3272     2121209      114.80        0.00          0.17   
34       978   26644   928683431      384.26     1114.59       5759.62   
35       979    1418     4673094      482.44     1257.80         29.15   
36       980   25905   553067364      560.31     1325.25       5793.51   
37       981   15960   312709097      398.98     1318.26       3264.53   
38       982   22801   642401705      334.28     1225.93       5079.78   
39       983    9929   435714604      188.70      795.73       2008.07   
40       984   27534   499776866      259.45     1066.20       3276.65   
41      1023   46979  1030623258      229.30      873.45       5473.44   
42      1024   18815   790782350      108.16     1053.26       3120.32   
43      1025    2503   118997955      184.08      841.82       1031.20   
44      1026   17834   125857565      128.56      507.91       1849.97   
45      1027   29495   338252511      130.01      497.35       2990.85   
46      

In [16]:
#to remove the rows with empty values as the empty values such as SUBCOUNTY_NAME, DISTRICT_NAME and Karamoja we can not assume redion and are a vital part of data
sorted_concat_df.dropna(inplace = True) 
sorted_concat_df

OBJECTID     POP        Area  S_Yield_Ha  M_Yield_Ha  Crop_Area_Ha  \
0        263   17244  1067176155      354.21     1137.47       7023.53   
1        264   52771   597575188      367.89     1163.00      13587.99   
2        265   27389    23972401      369.31     1167.01       1656.53   
3        266   38775   419111591      283.32      852.37       7087.82   
4        267   65704   880955930      373.84     1283.86      10398.25   
5        268   41273   652744859      271.19     1047.18      13278.52   
6        591   42221    60801942      160.59      650.12       1021.32   
7        592   38857    13071455      212.21     1229.21        175.08   
8        593   41454    67612362      208.04     1056.42        967.34   
9        594   99203   223116860      209.81      633.42       2834.35   
10       595   60070  1199409465      259.90      714.48        997.16   
11       596   33951   304557831      301.15     1030.06       2365.26   
12       597   56870   228457417      312.28     1125.69       6890.70   
13       598   37691  1742041261      298.45      871.61       3934.43   
14       599   22907   302572789      386.53      985.19       1890.25   
15       600   31820   397700845      164.65      815.26        401.80   
16       601  100919  1069331387      285.21     1273.40       2700.68   
17       602   16608  1330019236      284.18     1303.40       1491.77   
18       603   20394   309577387      295.73      765.33       1494.05   
19       604   24092   125330583      285.34     1396.99        957.49   
20       765   14377   188613057      501.26     1183.09        999.37   
21       766   11489   115941152      439.75     1061.15        788.46   
22       767   12573  1255760525      520.41     1259.12       1266.11   
23       768   18641   191684465      382.87     1003.30        698.17   
24       769   16834   250364895      407.50      962.96        925.05   
25       810   45015   658008801      193.90     1286.72       2513.27   
26       811   22541   371970302      200.70     1196.09       1539.40   
27       812   30181   596393139      239.05     1379.05       1652.00   
28       813    4053    17207110      146.12      861.21         60.77   
29       925   11772   277451997      311.01      546.14        145.24   
30       926   42115   596023780      133.67      591.27       2480.12   
31       927   21999   622266516      155.87      211.97       1111.66   
32       928    7161     2738777      131.98      241.54          0.32   
33       929    3272     2121209      114.80        0.00          0.17   
34       978   26644   928683431      384.26     1114.59       5759.62   
35       979    1418     4673094      482.44     1257.80         29.15   
36       980   25905   553067364      560.31     1325.25       5793.51   
37       981   15960   312709097      398.98     1318.26       3264.53   
38       982   22801   642401705      334.28     1225.93       5079.78   
39       983    9929   435714604      188.70      795.73       2008.07   
40       984   27534   499776866      259.45     1066.20       3276.65   
41      1023   46979  1030623258      229.30      873.45       5473.44   
42      1024   18815   790782350      108.16     1053.26       3120.32   
43      1025    2503   118997955      184.08      841.82       1031.20   
44      1026   17834   125857565      128.56      507.91       1849.97   
45      1027   29495   338252511      130.01      497.35       2990.85   
46      1028    8010   150906375      175.83      725.81        792.25   
47      1029   20790   158887552      134.08      528.70       1215.23   
48      1295   16470   769609687      329.76      779.23        792.90   
49      1313   23200  1794470536      120.86      748.83       6471.05   
50      1318   41493  2069554899      114.27      699.33       2217.29   
51      1320   16588   839293722      430.56     1248.96       1161.39   

    S_Area_Ha  M_Area_Ha  S_Prod_Tot  M_Prod_Tot              SUBCOUNTY_NAME  \
0

### getting information to describe the data

In [17]:
#describing the datadrame
sorted_concat_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52 entries, 0 to 51
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   OBJECTID        52 non-null     int64  
 1   POP             52 non-null     int64  
 2   Area            52 non-null     int64  
 3   S_Yield_Ha      52 non-null     float64
 4   M_Yield_Ha      52 non-null     float64
 5   Crop_Area_Ha    52 non-null     float64
 6   S_Area_Ha       52 non-null     float64
 7   M_Area_Ha       52 non-null     float64
 8   S_Prod_Tot      52 non-null     float64
 9   M_Prod_Tot      52 non-null     float64
 10  SUBCOUNTY_NAME  52 non-null     object 
 11  DISTRICT_NAME   52 non-null     object 
 12  Karamoja        52 non-null     object 
dtypes: float64(7), int64(3), object(3)
memory usage: 5.7+ KB


In [18]:
#information on the dataframe
sorted_concat_df.describe()

OBJECTID            POP          Area  S_Yield_Ha   M_Yield_Ha  \
count    52.000000      52.000000  5.200000e+01   52.000000    52.000000   
mean    787.865385   28934.692308  5.331913e+08  274.165577   940.259423   
std     280.101314   20865.122974  4.913308e+08  118.569529   321.641686   
min     263.000000    1418.000000  2.121209e+06  108.160000     0.000000   
25%     597.750000   16558.500000  1.568923e+08  173.035000   743.075000   
50%     810.500000   23053.500000  3.848356e+08  277.255000  1016.680000   
75%     982.250000   39461.000000  7.749029e+08  368.245000  1203.550000   
max    1320.000000  100919.000000  2.069555e+09  560.310000  1396.990000   

       Crop_Area_Ha     S_Area_Ha    M_Area_Ha    S_Prod_Tot    M_Prod_Tot  
count     52.000000     52.000000    52.000000  5.200000e+01  5.200000e+01  
mean    2839.646731   2253.143269   536.300577  6.557443e+05  5.500730e+05  
std     3110.505842   2954.355752   724.092303  9.915839e+05  7.939707e+05  
min        0.170000      0.130000     0.000000  1.728000e+01  0.000000e+00  
25%      964.877500    405.397500    79.825000  1.210555e+05  6.087012e+04  
50%     1654.265000   1231.825000   326.480000  2.543687e+05  2.896239e+05  
75%     3267.560000   2429.985000   740.297500  6.040942e+05  8.114574e+05  
max    13587.990000  12964.500000  3840.700000  4.582294e+06  4.365058e+06

# Explanation of data.
### is there a cooorelation btw population and crop area hactre?

In [19]:
import numpy as np
import pandas as pd
from scipy.stats import spearmanr

# Calculate Spearman's rank correlation coefficient and p-value
correlation, p_value = spearmanr(sorted_concat_df['Area'], sorted_concat_df['Crop_Area_Ha'])

# Print the results
print(f"Spearman's rank correlation coefficient: {correlation:.4f}")
print(f"P-value: {p_value:.4f}")


Spearman's rank correlation coefficient: 0.5986
P-value: 0.0000


there is an average of 3110.5 Ha available for farming crops. on average 724.0 Ha is used for growing Maize and 2253.1 is used for growing sorghum.on average a yeild of sorghum and maize are 274.17	940.26 kg's accordingly.but the highest harvest of sorgum and maize is 560.31	1396.99	accordingly.
the Spearman's rank correlation coefficient is 0.5986, No monotonic relationship.Area does not affect the farm land available for planting crops.

In [20]:
#saving the file on my local device as excel file
#sorted_concat_df.to_excel(r"C:\Users\ericg\Downloads\DATA\cleaned dataframe.xlsx", index=False)

### link to tableau public
https://public.tableau.com/app/profile/eric.giitwa/viz/TABLEAU_PROJECT_2_UGANDA/PROJECTDASHBOARD?publish=yes

### link to powerpoint presentation  
https://docs.google.com/presentation/d/1Dco2YJNAAWrFJt3hG5W9EcPKSCcd35qcQswNnPV8ZIw/edit?usp=sharing